### PR TITLE
The One Probe integration. Improve WAILA integration.

### PR DESCRIPTION
**Pull request by elifoster**
WAILA integration and The One Probe integration are 1:1; what was already shown in the WAILA tooltip is shown in The One Probe tooltip.
Remove Bed WAILA integration and replace it with properly overriding getPickBlock in BlockBedGeneric.

Here is an example of what it looks like: 

Open and empty basin:
Open and empty basin

Closed and full basin:
Closed and full basin

Potential improvements I can think of:
- Improve tooltip formatting by adding formatting codes to the localizations, rather than the code. For example, `toolTip.add(String.format("%s : %d%%", LangHelper.translate("waila.composter.composting"), (int) (tag.getFloat("Compost") * 100f)));` would become `toolTip.add(LangHelper.translate("waila.composter.composting", (int) (tag.getFloat("Compost") * 100f));` and likewise for the The One Probe equivalent code.
- Centralize the code for generating the WAILA/The One Probe tooltips so that the code is in one location rather than duplicated in two. I'm not certain how we could do this effectively since the way WAILA and The One Probe work are entirely different (specifically, TOP uses chained calls, while WAILA uses a list).

### CODE REVIEW

File: java/squeek/veganoption/blocks/BlockComposter.java
Comment (squeek502) [inline java/squeek/veganoption/blocks/BlockComposter.java]: This `ProviderComposter.DEGREE_SYMBOL` reference might cause problems when Waila is not installed (ProviderComposter is normally only loaded when Waila is installed, so the Waila interface is assumed to exist).

Could just move the DEGREE_SYMBOL constant to BlockComposter and use `BlockComposter.DEGREE_SYMBOL` in the Waila file instead.

**Review by squeek502 — CHANGES_REQUESTED** [review]

**Comment by squeek502** [comment]
I generally try to avoid adding full mods to the dev environment whenever possible (to better test that they are optional and to keep the default environment lean), but it looks like that's not really possible with TOP, unfortunately. Perhaps we could request API-only packages be added to the TOP maven.

EDIT: See https://github.com/McJty/TheOneProbe/issues/66

**Comment by elifoster** [comment]
I've moved `DEGREE_SYMBOL` so that the TOP integration is not dependent on the WAILA integration classes.

I agree with the API package stuff. Are we going to wait until McJty/TheOneProbe#66 is resolved to merge this, or simply update the buildscript once that issue gets resolved?

**Review by squeek502 — APPROVED** [review]

**Comment by squeek502** [comment]
Looks good, we can merge it now.

**Comment by elifoster** [comment]
What do you think of the two potential improvements I listed above?

**Comment by squeek502** [comment]
1. I think this change would make sense, as it'd also allow for the [% sign placement to be based on localization](https://en.wikipedia.org/wiki/Percent_sign#Correct_style)
2. This one might be more difficult. I was thinking we could add something to the Block classes like `getContextStrings` that would return a `List<String>` and then the TOP methods could just loop through that and chain `text` calls, but Waila depends on NBT syncing that might make that method not quite work without special handling ([for example](https://github.com/squeek502/VeganOption/blob/1.10.2/java/squeek/veganoption/integration/waila/ProviderComposter.java#L37)).

**Pull request merged**
